python/pyPamtra/importer.py

```diff
@@ -3062,3 +3062,993 @@ def ncToDict(ncFilePath,keys='all',joinDimension='time',offsetKeys={},ncLib='net
+    sonde_dict[key], sonde_ipflag[key] = fill_gaps(sonde_dict[key], verbose=verbose)
+
+  # # # # # # # # # # # # # # # # # # # # # # # # # #
+
+  # Check for remaining NaNs in meteorological variables:
+  # RH, T and P (and the height coordinate Z):
+  if (np.any([np.isnan(sonde_dict['T']), np.isnan(sonde_dict['P']), np.isnan(sonde_dict['RH']),
+    np.isnan(sonde_dict['Z'])])):
+    if verbose:
+      print('     NaN-counts: %d, %d, %d, %d' % (
+        np.isnan(sonde_dict['T']).sum(),
+        np.isnan(sonde_dict['P']).sum(),
+        np.isnan(sonde_dict['RH']).sum(),
+        np.isnan(sonde_dict['Z']).sum()))
+    raise ValueError("Nans still remained in T, P, RH or Z. Aborting execution of dropsonde %s. Please contact 'a.walbroel@uni-koeln.de'."
+      % file_raw_sondes)
+
+  # Check if surface winds are not nan: (index [1] is used because [0] is at 0 m above sea level)
+  sfc_wind_available = ~np.isnan(sonde_dict['u_wind'][1] + sonde_dict['v_wind'][1])
+  if not sfc_wind_available:
+    if verbose:
+      print("WARNING, NaN in u_wind[1] or v_wind[1]. Surface winds won't be included in PAMTRA for %s !"%(
+        file_raw_sondes))
+      print('     NaN-counts (u,v): %d, %d' % (
+        np.isnan(sonde_dict['u_wind'][1]),
+        np.isnan(sonde_dict['v_wind'][1])))
+
+
+  # Create pamtra object; change settings:
+  pam = pyPamtra()
+
+  pam.nmlSet['passive'] = True						# passive simulation
+  pam.nmlSet['active'] = False						# False: no radar simulation
+
+  # Create dictionary for PAMTRA data:
+  pamData = dict()
+
+  Nh = len(sonde_dict['Z'])		# number of height levels
+  Nx = 1						# number of sondes (is always 1 because one sonde is considered at a time)
+  nhydros = 0					# number of hydrometeors (here: clear sky assumed)
+  shape2d = [Nx, Nx]
+  shape3d = shape2d + [Nh]
+
+  pamData['hgt_lev'] = np.broadcast_to(sonde_dict['Z'], shape3d)	  # height
+  pamData['press_lev'] = np.broadcast_to(sonde_dict['P'][:], shape3d)	  # pressure
+  pamData['temp_lev'] = np.broadcast_to(sonde_dict['T'][:], shape3d)	  # temperature
+  pamData['relhum_lev'] = np.broadcast_to(sonde_dict['RH'][:], shape3d)  # relative humidity
+
+  if sfc_wind_available:
+    pamData['wind10u'] = np.broadcast_to(sonde_dict['u_wind'][1], shape2d)
+    pamData['wind10v'] = np.broadcast_to(sonde_dict['v_wind'][1], shape2d)
+
+  if not np.any([np.isnan(sonde_dict['reference_lon']), np.isnan(sonde_dict['reference_lat'])]):
+    pamData['lon'] = np.broadcast_to(sonde_dict['reference_lon'], shape2d)
+    pamData['lat'] = np.broadcast_to(sonde_dict['reference_lat'], shape2d)
+
+  else:		# use highest non nan lat/lon values:
+    pamData['lon'] = np.broadcast_to(sonde_dict['lon'][~np.isnan(sonde_dict['lon'])][-1], shape2d)
+    pamData['lat'] = np.broadcast_to(sonde_dict['lat'][~np.isnan(sonde_dict['lat'])][-1], shape2d)
+
+  pamData['timestamp'] = np.broadcast_to(sonde_dict['launch_time'], shape2d)
+  pamData['groundtemp'] = np.broadcast_to(sonde_dict['T'][0], shape2d)
```

Comment:
One more comment: I would prefer the groundtemp to be optionally passed as argument. You assume here, that air temperature at 15 or 30 m above the surface is in equilibrium with the sea. That is not always the case. (e.g. in the Arctic). I think more flexibility here would be great.

Reply:
sonde_dict['T'][0] is not at 15 or 30 m but a linearly extrapolated value from the lowest dropsonde measurements. Anyway, it certainly does not equal the sea surface temperature.

This would require additional input information to the importer function. I'll add a sst=np.nan argument to the function read_AVAPS_dropsonde(...).